File: lib/models/memory.dart
class Memory {
  final String id;
  final String title;
  final String notes;
  final DateTime date;
  final List<String> media;
  final String? url;

  Memory({
    required this.id,
    required this.title,
    required this.notes,
    required this.date,
    required this.media,
    this.url,
  });

  factory Memory.fromJson(Map<String, dynamic> json) {
    return Memory(
      id: json['_id'],
      title: json['title'],
      notes: json['notes'],
      date: DateTime.parse(json['date']),
      media: List<String>.from(json['media']),
      url: json['url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'notes': notes,
      'date': date.toIso8601String(),
      'media': media,
    };
  }
}

File: backend/models/memory.js
const mongoose = require('mongoose');

const MemorySchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Please add a title'],
    trim: true,
  },
  media: [
    {
      type: String,
      required: true,
    },
  ],
  date: {
    type: Date,
    required: true,
  },
  notes: {
    type: String,
    required: true,
  },
  url: {
    type: String,
    required: true,
    unique: true,
  },
  user: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Memory', MemorySchema);
